pkg/volume/csi/csi_plugin.go

```diff
@@ -603,14 +614,17 @@ func (p *csiPlugin) CanAttach(spec *volume.Spec) (bool, error) {
 
 // CanDeviceMount returns true if the spec supports device mount
 func (p *csiPlugin) CanDeviceMount(spec *volume.Spec) (bool, error) {
-	driverMode, err := p.getDriverMode(spec)
-	if err != nil {
-		return false, err
-	}
+	inlineEnabled := utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume)
+	if inlineEnabled {
+		csiVolumeMode, err := p.getCSIVolumeMode(spec)
+		if err != nil {
+			return false, err
+		}
 
-	if driverMode == ephemeralDriverMode {
-		klog.V(5).Info(log("plugin.CanDeviceMount skipped ephemeral mode detected for spec %v", spec.Name()))
-		return false, nil
+		if csiVolumeMode == ephemeralVolumeMode {
+			klog.V(5).Info(log("plugin.CanDeviceMount skipped ephemeral mode detected for spec %v", spec.Name()))
+			return false, nil
+		}
```

Comment:
Would this make it more clear?

If you do not think so, ignore it.

```
if !inlineEnabled {
    return true, nil
}
driverMode, err := p.getDriverMode(spec)
...

return true, nil

```

Reply:
Yes, I think that's better and more consistent with usual Go style. I changed it, please have another look.